### DISCUSSION TITLE
Separate client and server?

### DISCUSSION THREAD

**Discussion by danirukun** [imartinez/privateGPT#596] [Ideas]
I want to be able to run PrivateGPT on a shared machine, and make it accessible to multiple users through e.g. some kind of API. Maybe PrivateGPT.py should be split up a bit into a client and server?

**Comment by ROHNB** [comment]
I'm interested in the same. It would be great if we could operate PrivateGPT on a communal machine, allowing multiple users to interact with it, perhaps via an API interface. Additionally, I'm interested in understanding if a LocalAI API support could be leveraged for the communication with PrivateGPT. Could such a solution be feasible and efficient?

**Reply by douglasg14b** [reply]
100%. Stuff this onto a beefy server with fast storage, and my other devices can interact with it. That would make this fantastic.

Begin able to leverage LocalAI would really raise this up.

**Reply by kon75** [reply]
interested in this as well!
